src/test/java/org/sqlite/ConnectionTest.java

```diff
@@ -409,4 +415,46 @@ public void overrideURIPragmaValuesWithProperties() throws Exception {
         stat.close();
         conn.close();
     }
+
+    @SuppressWarnings("resource")
+    @Test
+    public void openNonExistingFileNoCreate() {
+        Path nonExisting = Paths.get("non_existing.db").toAbsolutePath();
+        assertThat(Files.exists(nonExisting)).isFalse();
+        SQLiteConfig cfg = new SQLiteConfig();
+        cfg.resetOpenMode(SQLiteOpenMode.CREATE);
+        assertThatExceptionOfType(SQLiteException.class)
+                .isThrownBy(() -> cfg.createConnection("jdbc:sqlite:" + nonExisting))
+                .satisfies(
+                        e ->
+                                assertThat(e.getResultCode())
+                                        .isEqualTo(SQLiteErrorCode.SQLITE_CANTOPEN));
+        assertThat(Files.exists(nonExisting)).isFalse();
+    }
+
+    @SuppressWarnings("resource")
```

Comment:
why do you need this ?

Reply:
I'm developing with Intellij and get this warning 
![image](https://github.com/user-attachments/assets/661a6b62-0b43-4ced-9aac-46e9f707d354)
Intellij also offers me to add createConnection to some list of ignored methods (not sure where that ends up). The annotation I used should also be respected by eclipse, so is a little bit more cross-IDE.
I've found a way to put it closer to the culprit. The annotation only works on fields, methods or statements which is why it needs introduction of an unused variable though ..